databricks/koalas/series.py

```diff
@@ -3184,6 +3184,91 @@ def copy(self) -> 'Series':
         """
         return _col(DataFrame(self._internal.copy()))
 
+    def mode(self, dropna=True) -> 'Series':
+        """
+        Return the mode(s) of the dataset.
+
+        Always returns Series even if only one value is returned.
+
+        Parameters
+        ----------
+        dropna : bool, default True
+            Don't consider counts of NaN/NaT.
+
+        Returns
+        -------
+        Series
+            Modes of the Series in sorted order.
+
+        Examples
+        --------
+        >>> s = ks.Series([0, 0, 1, 1, 1, np.nan, np.nan, np.nan])
+        >>> s
+        0    0.0
+        1    0.0
+        2    1.0
+        3    1.0
+        4    1.0
+        5    NaN
+        6    NaN
+        7    NaN
+        Name: 0, dtype: float64
+
+        >>> s.mode()
+        0    1.0
+        Name: 0, dtype: float64
+
+        If there are several same modes, all items are shown
+
+        >>> s = ks.Series([0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3,
+        ...                np.nan, np.nan, np.nan])
+        >>> s
+        0     0.0
+        1     0.0
+        2     1.0
+        3     1.0
+        4     1.0
+        5     2.0
+        6     2.0
+        7     2.0
+        8     3.0
+        9     3.0
+        10    3.0
+        11    NaN
+        12    NaN
+        13    NaN
+        Name: 0, dtype: float64
+
+        >>> s.mode()
+        0    1.0
+        1    2.0
+        2    3.0
+        Name: 0, dtype: float64
+
+        With 'dropna' set to 'False', we can also see NaN in the result
+
+        >>> s.mode(False)
+        0    NaN
+        1    1.0
+        2    2.0
+        3    3.0
+        Name: 0, dtype: float64
+        """
+        ser_count = self.value_counts(dropna=dropna)
+        sdf_count = ser_count._internal.sdf
+        most_value = sdf_count.head(1)[0][1]
+        sdf_most_value = (sdf_count.where("count == {}".format(most_value))
+                                   .sort('__index_level_0__')
+                                   .rdd.zipWithIndex().toDF())
```

Comment:
What's this doing?

Reply:
Thanks for commenting and sorry for confusing.

let me explain these with some examples below.

first, we have series `ser`.
```python
>>> ser
lama    speed      45.0
        weight    250.0
        length      1.0
cow     speed      30.0
        weight    250.0
        length      1.0
falcon  speed     250.0
        weight      1.0
        length      0.3
Name: 0, dtype: float64
```

`sdf_count` is literally spark data frame that has count of each values

`ser`'s `sdf_count` looks like

```
>>> sdf_count.show()
+-----------------+-----+
|__index_level_0__|count|
+-----------------+-----+
|              1.0|    3|
|            250.0|    3|
|             30.0|    1|
|             45.0|    1|
|              0.3|    1|
+-----------------+-----+
```
and `most_value` is the mode value of `ser`.

in this case, it's 3

```
>>> most_value
3
```

And i used zipWithIndex() to make a column that have ascending number from 0.

because if there are plural mode values,

pandas shows them in ascending order with ascending number from 0 like below.

```python
>>> s.to_pandas().mode()
0      1.0
1    250.0
dtype: float64
```

and `sdf_most_value` looks like:

```
>>> sdf_most_value.show()
+----------+---+
|        _1| _2|
+----------+---+
|  [1.0, 3]|  0|
|[250.0, 3]|  1|
+----------+---+
```

so finally, i make `sdf` using data from `sdf_most_value`

```
>>> sdf.show()
+---+-----+
| _2|    0|
+---+-----+
|  0|  1.0|
|  1|250.0|
+---+-----+
```

actually while i'm doing this, I thought this code might be a bit harder to understand for someone.

However, there are no other idea came up from my brain, and sorry to make confuse you with complicated code. 😢

Reply:
Koalas automatically handle adding index if emtpy. We just need to set up the `sdf_most_value` without index.

Reply:
@ueshin Thanks!! i'm going to fix them. :)